source/common/config/grpc_subscription_impl.cc

```diff
@@ -67,6 +67,17 @@ void GrpcSubscriptionImpl::onConfigUpdate(
             resources.size(), version_info);
 }
 
+void GrpcSubscriptionImpl::onConfigUpdate(
+    const Protobuf::RepeatedPtrField<envoy::service::discovery::v3::Resource>& added_resources,
+    const Protobuf::RepeatedPtrField<std::string>& removed_resources,
+    const std::string& system_version_info) {
+  disableInitFetchTimeoutTimer();
+  stats_.update_attempt_.inc();
+  callbacks_.onConfigUpdate(added_resources, removed_resources, system_version_info);
+  stats_.update_success_.inc();
+  stats_.version_.set(HashUtil::xxHash64(system_version_info));
+}
+
```

Comment:
Sorry can you briefly summarize why we have this new code here if we are just removing a redundant interface? Is it possible to somehow have a common update helper that does the logging, version setting, stats, etc. as the function above so we make sure we have consistent behavior moving forward?

Reply:
I removed this code and replaced it with an assert. The issue is that we used to have 2 different callback interfaces, one for sotw and one for delta. Now they're merged. This method is the delta method. We also have 2 different classes that implement this interface. GrpcSubscriptionImpl is the sotw implementation. It should never receive this callback. The next step will be to merge the DeltaSubscriptionImpl and GrpcSubscriptionImpl classes at which time this code will be needed. I think that adding the assert here is the easiest way to move forward without ending up with a massive change. It should be removed in my next PR.